ffi/src/x509.rs

```diff
@@ -215,4 +215,188 @@ pub mod ffi {
             Ok(Box::new(Cert(cert)))
         }
     }
+
+    #[diplomat::opaque]
+    pub struct CertVec(pub Vec<certificate::Cert>);
+
+    impl CertVec {
+        pub fn next(&mut self) -> Option<Box<Cert>> {
+            self.0.pop().map(|cert| Box::new(Cert(cert)))
+        }
+    }
```

Comment:
suggestion: Given the API and semantic, you can call that `CertIterator` instead of `Vec`

Reply:
updated